### PR TITLE
afw tickets/DM-2535: Remove compound fields from afw::table

### PR DESCRIPTION
**Pull request by TallJimbo**
This PR removes all usage of compound fields from afw (there are similar changes in other packages), modifies the FITS read code to convert saved compound fields into the corresponding `FunctorKeys` that replace them, and finally removes support for the compound fields themselves.

### CODE REVIEW

File: include/lsst/afw/math/KernelPersistenceHelper.h [hunk 1]
```diff
@@ -29,14 +29,15 @@
 #include "lsst/afw/table/io/OutputArchive.h"
 #include "lsst/afw/table/io/InputArchive.h"
 #include "lsst/afw/table/io/CatalogVector.h"
+#include "lsst/afw/table/aggregates.h"
```
Comment (r-owen) [inline include/lsst/afw/math/KernelPersistenceHelper.h]: include table.h?

File: include/lsst/afw/table/Simple.h [hunk 2]
```diff
@@ -205,7 +206,7 @@ inline void SimpleRecord::setId(RecordId id) { set(SimpleTable::getIdKey(), id);
 
 inline IcrsCoord SimpleRecord::getCoord() const { return get(SimpleTable::getCoordKey()); }
 inline void SimpleRecord::setCoord(IcrsCoord const & coord) { set(SimpleTable::getCoordKey(), coord); }
-inline void SimpleRecord::setCoord(Coord const & coord) { set(SimpleTable::getCoordKey(), coord); }
+inline void SimpleRecord::setCoord(Coord const & coord) { SimpleTable::getCoordKey().set(*this, coord); }
```
Comment (r-owen) [inline include/lsst/afw/table/Simple.h]: Does this convert coord to an IcrsCoord? I don't see it in the code. If it's not happening, it should (though I realize it's a pre-existing issue). If it is happening automagically then please document it.

File: include/lsst/afw/math/KernelPersistenceHelper.h (hunk 1 above)
Comment (TallJimbo) [inline include/lsst/afw/math/KernelPersistenceHelper.h]: table.h includes a lot more that we don't need.

File: include/lsst/afw/table/Simple.h (hunk 2 above)
Comment (TallJimbo) [inline include/lsst/afw/table/Simple.h]: This is a preexisting bug I discovered on this issue and worked around but should have reported.  It's now https://jira.lsstcorp.org/browse/DM-2872.

**Comment by TallJimbo** [comment]
Merged to master on tickets/DM-1766.

**Pull request closed without merging**